src/components/organisms/ConceptPicker.tsx

```diff
@@ -0,0 +1,167 @@
+import { useEffect, useRef, useState } from "react";
+import { NextRouter, useRouter } from "next/router";
+
+import { Label } from "@/components/labels/Label";
+import { InputCheck } from "@/components/forms/Checkbox";
+import { Select } from "@/components/atoms/select/Select";
+
+import { fetchAndProcessConcepts } from "@/utils/processConcepts";
+import { groupByRootConcept } from "@/utils/conceptsGroupedbyRootConcept";
+
+import { QUERY_PARAMS } from "@/constants/queryParams";
+
+import { TConcept } from "@/types";
+
+type TProps = {
+  concepts: TConcept[];
+  startingSort?: TSort;
+};
+
+type TSort = "A-Z" | "Grouped";
+
+const SORT_OPTIONS: TSort[] = ["A-Z", "Grouped"];
+
+const isSelected = (queryValue: string | string[] | undefined, option: string) => {
+  if (!queryValue) {
+    return false;
+  }
+  if (typeof queryValue === "string") {
+    return queryValue === option;
+  } else {
+    return queryValue.includes(option);
+  }
+};
+
+const conceptsSorter = (a: TConcept, b: TConcept, sort: TSort) => {
+  if (sort === "A-Z") {
+    return a.preferred_label.localeCompare(b.preferred_label);
+  } else if (sort === "Grouped") {
+    return a.recursive_subconcept_of[0].localeCompare(b.recursive_subconcept_of[0]);
+  }
+  return 0;
+};
+
+const filterConcepts = (concepts: TConcept[], search: string) => {
+  return concepts.filter(
+    (concept) =>
+      concept.preferred_label.toLowerCase().includes(search.toLowerCase()) ||
+      concept.alternative_labels.some((label) => label.toLowerCase().includes(search.toLowerCase()))
+  );
+};
+
+const onConceptChange = (router: NextRouter, concept: TConcept) => {
+  const query = { ...router.query };
+  let selectedConcepts: string | string[] = query[QUERY_PARAMS.concept_name] || [];
```

Comment:
Suggestion: You can probably save yourself some if statements here by getting the concept params into a guaranteed array. I'd use something like `[query[QUERY_PARAMS.concept_name]].flatten()`.

Reply:
Still needed to check we had a value in the query else I was getting an initial array of `[undefined]` but good shout using `flat()` makes things easier afterwards.